### ISSUE TITLE
mGBA 3DS Interface

### ISSUE THREAD

**Issue by Leafgreen26** [mgba-emu/mgba#981]
Is it possible that for the 3DS, the interface could look something more like SNES9X_3DS or VirtuaNES? Concept:

THIS POST IS JUST A COLLECTION OF CONCEPT RENDERINGS. THERE IS NO BUILD OR CODE AT THIS TIME.

I've been mulling over the idea of implementing bubble2k16k's emulator GUI into mGBA. I'm at the preliminary stage of checking for feasibility and practicality. Is it within my means and capability? Here are some mock up pictures of what I envision. The purple based color scheme was borrowed from https://mgba.io/ , to keep her emulator distinct from bubble2k16's signature blue.
![image](https://user-images.githubusercontent.com/35386457/35636831-04635670-0678-11e8-8222-626e9dc2ee63.png)

Edit - The last panel with the exiting 'Leaving so soon?' swaps out the red for Bubble2k16's main blue, kinda like a tribute for where the derived GUI design stems from.

**Comment by Leafgreen26** [comment]
![image](https://user-images.githubusercontent.com/35386457/35636855-12075cae-0678-11e8-821f-2c1ad98d94df.png)

![image](https://user-images.githubusercontent.com/35386457/35636861-1ac93cf4-0678-11e8-95c4-516241473980.png)

**Comment by endrift** [comment]
Yeah, I got the email, but hadn't had a chance to reply yet. While I do really like this design, I'd rather give it my own spin. It'll be a bunch of work but I'm actually looking into starting on that right now.

**Comment by Leafgreen26** [comment]
![image](https://user-images.githubusercontent.com/35386457/35636880-2748be00-0678-11e8-8a94-6426a96550d3.png)
![image](https://user-images.githubusercontent.com/35386457/35636886-2ab8e998-0678-11e8-89ee-477a9034bf1c.png)
![image](https://user-images.githubusercontent.com/35386457/35636897-2fb3c8d2-0678-11e8-8a18-203ec2a630b2.png)
![image](https://user-images.githubusercontent.com/35386457/35636910-35ded1d4-0678-11e8-8633-13c655eb35a1.png)
![image](https://user-images.githubusercontent.com/35386457/35636917-3a52e444-0678-11e8-9a48-f6d91180e70f.png)

**Comment by Leafgreen26** [comment]
Cool. Sorry if I was being spammy, I just wanted to make sure you got it. I didn't know if the email listed was current or anything.

**Comment by endrift** [comment]
Yep, I got it, and that's the most up-to-date email for me. Thanks for pushing me on this since it's been a long time coming.

**Comment by endrift** [comment]
See also #805.

**Comment by TurdPooCharger** [comment]
I am the artist of those renderings. I was coming to github to look for and report a bug I found on a v0.7.0 prebuild and saw this issue posted. The rest of those mockups are posted [here](https://gbatemp.net/threads/psa-the-lastest-mgba-nightly-runs-pretty-fast.485069/page-8#post-7797525).

A long time ago, I (think?) I remember lurking on GBAtemp and reading about a another user requesting from either you (or Jeffrey Pfau) to implement a GUI similar to bubble2k16's, but that such a feature was not high-up on priority because function/emulation performance took precedent over form at the time and so-and-so user was free to do it himself. He was persistent with his request to the point where your/Jeff's patience was tested.

If you guys have ever used one of bubble's 3DS emulators, his are set up so that only roms with certain file associations are visible. For mGBA, this would mean only files ending in .gb, .gbc, .gba, .bin, and .png (if a screenshot viewer is ever implemented) are the only ones recognized.

I've been studying both yours and his source codes in trying to splice a reworked version of his GUI onto your emulator but only got past "Hello World" and 3DS examples level of understanding. So far from my understanding, his GUI calls up specialized, recreated fonts not shared from the 3ds system font. If you end up reverse engineering/borrowing his codes, just know that GUI is a piece.

**Comment by Leafgreen26** [comment]
For the record, to clear any confusion that might have been made, I was not taking credit for the mock-ups above, I just copied the whole post and wanted the devs to see it.

**Comment by TurdPooCharger** [comment]
No accusation coming from me. You're safe, buddy. :)

I didn't want to hassle or directly showcase this to her (and other devs) as they already have enough on their plate. This feature is nonessential from a design standpoint and doesn't fix what's not broken. With no working demonstration code, mock-ups might as well be treated as vaporware.
